pkg/kubelet/dockertools/exec.go

```diff
@@ -121,6 +126,14 @@ func (*NativeExecHandler) ExecInContainer(client DockerInterface, container *doc
 	if err != nil {
 		return err
 	}
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
+			// FIXME: we should kill the process in the container,
+			// but I couldn't find anything in the Docker API docs
+			// on how to do it with the *Exec APIs.
```

Comment:
It's impossible to use the Docker API to stop or kill an Exec session, unfortunately. And they have stated they don't intend to add support to do so.